src/systemtest/java/com/ericsson/ei/systemtest/utils/StepsUtils.java

```diff
@@ -26,18 +26,21 @@
     * @param jenkinsToken - Token to the jenkins job.
     *
     * @return boolean - If the creation was a success or not
-    *
-    * @throws JSONException
-    * @throws URISyntaxException
-    * @throws IOException
+     * @throws Exception
     */
-    public static boolean createJenkinsJob(String jenkinsJobName, String scriptFileName, String jenkinsBaseUrl, String jenkinsUsername, String jenkinsPassword, String jenkinsToken, String remremBaseUrl) throws URISyntaxException, JSONException, IOException {
+    public static boolean createJenkinsJob(String jenkinsJobName, String scriptFileName, String jenkinsBaseUrl, String jenkinsUsername, String jenkinsPassword, String jenkinsToken, String jenkinsJobXml, String remremBaseUrl) throws Exception {
         jenkinsManager = new JenkinsManager(jenkinsBaseUrl, jenkinsUsername, jenkinsPassword);
+
+        if(!jenkinsManager.pluginExists("Groovy")) {
+            jenkinsManager.installPlugin("Groovy", "2.1");
+            jenkinsManager.restartJenkins();
+        }
         String script = new String(Files.readAllBytes(Paths.get(scriptFileName)));
         script = script.replace("<REMREM_BASE_URL>", remremBaseUrl);
 
-        String xmlJobData = jenkinsManager.getXmlJobData(jenkinsToken, script);
-        return jenkinsManager.createJob(jenkinsJobName, xmlJobData);
+        String xmlJobData = Utils.getResourceFileAsString(jenkinsJobXml);
+        xmlJobData = xmlJobData.replace("${shell.script}", script);
```

Comment:
Is ${shell.script} variable defined anywhere? If not, I think replace string varariable format should be the same in all situations and don't mix format.

Reply:
Agreed, ill change the formats